### PR TITLE
[lld-macho] Make ObjC category checker print the source file name of …

### PR DESCRIPTION
**Pull request by alx32**
When printing category conflicts in the ObjC category checker, also print the source file name of the problematic categories. Currently we only print the object file name. This change is mostly useful only for thinLTO builds, where the object file name will be of form 999.arm64.lto.o and thus does not reveal any information about the original source file.

### CODE REVIEW

**Comment by kyulee-com** [comment]
> This change is mostly useful only for thinLTO builds

I'm curious how it looks like with (full)lto builds where bitcodes are merged.

**Comment by alx32** [comment]
> I'm curious how it looks like with (full)lto builds where bitcodes are merged.

Good catch. In that case the file paths won't be valid. i.e it'll be something like:
> defined in category TheCategory from /tmp/lto.tmp (RandomFile.mm)

However, I don't see an easy way to tell at this location if we're within the LTO scenario.

File: lld/test/MachO/objc-category-conflicts.s [hunk 1]
```diff
@@ -14,6 +17,14 @@
 # RUN: %no-fatal-warnings-lld --check-category-conflicts -dylib -lobjc %t/libklass.dylib %t/cat1.o \
 # RUN:   %t/cat2.o -o /dev/null 2>&1 | FileCheck %s --check-prefix=CATCAT
 
+# RUN: %no-fatal-warnings-lld --check-category-conflicts -dylib -lobjc %t/klass_w_sym.o %t/cat1_w_sym.o %t/cat2_w_sym.o -o \
+# RUN:   /dev/null 2>&1 | FileCheck %s --check-prefixes=CATCLS_W_SYM,CATCAT_W_SYM
+# RUN: %no-fatal-warnings-lld --check-category-conflicts -dylib -lobjc %t/libklass.dylib %t/cat1_w_sym.o \
+# RUN:   %t/cat2_w_sym.o -o /dev/null 2>&1 | FileCheck %s --check-prefix=CATCAT_W_SYM
+
+# RUN: %no-fatal-warnings-lld --check-category-conflicts -dylib -lobjc %t/klass.o %t/cat1.o %t/cat2.o -o \
+# RUN:   /dev/null 2>&1 | FileCheck %s --check-prefixes=CATCLS,CATCAT
```
Comment (thevinster) [inline lld/test/MachO/objc-category-conflicts.s:26]: This looks like it's doing the same thing on L15? Is there a reason for duplicating this?

**Review by thevinster — APPROVED** [review]
LGTM, just a minor nit.

> However, I don't see an easy way to tell at this location if we're within the LTO scenario.

That's unfortunate. Spitting out some RandomFile might lead someone on a wild goose chase and I would prefer (if possible) to not even emit the source file in that scenario. In any case, this is already a quality improvement.

**Comment by kyulee-com** [comment]
lgtm

**Review by kyulee-com — APPROVED** [review]

**Pull request merged**